### ISSUE TITLE
Regression: variables in "for" loops are handled incorrectly when modified during the loop

### ISSUE THREAD

**Issue by seanlip** [skulpt/skulpt#749]
In the latest version of Skulpt ([a472e2e](https://github.com/skulpt/skulpt/tree/a472e2e170302c123db35e366cbd917b25a55998)), the following code results in the value of `counter` being set to `4`, whereas it should only end up being `2` because the iterated-over list is being modified during the iteration.

```
L = ['a', 'b', 'c', 'd']
counter = 0
for c in L:
    L.pop()
    counter += 1
print counter
```

Note that this works correctly on www.skulpt.org, which runs an earlier version of Skulpt (not sure which one), so I think it's a regression in the library. I did a little digging and it looks like it was working correctly in v0.10.0 (b431c9c) but is broken in c140150, so it's probably due to some commit between those two boundaries.

**Comment by albertjan** [comment]
Hey sorry for the slow response. Good catch, and interesting that this functionality changed.

**Comment by prescod** [comment]
The deeper problem seems to be with iterators:

L = [1,2,3,4]
i = iter(L)
L.pop()
count = len(list(i))
print(count)

**Comment by albertjan** [comment]
yeah I think I figured it out. 😄 we make a copy of the list when we create a list iterator.

**Comment by prescod** [comment]
@albertjan : So here's the thing: I have a very simple solution to the problem.

diff --git a/src/list.js b/src/list.js
index 1ea9753..da5b1dc 100644
--- a/src/list.js
+++ b/src/list.js
@@ -655,11 +655,10 @@ Sk.builtin.list_iter_ = function (lst) {
         return new Sk.builtin.list_iter_(lst);
     }
     this.$index = 0;
-    this.lst = lst.v.slice();
-    this.sq$length = this.lst.length;
+    this.lst = lst.v;
     this.tp$iter = this;
     this.tp$iternext = function () {
-        if (this.$index >= this.sq$length) {
+        if (this.$index >= this.lst.length) {
             return undefined;
         }
         return this.lst[this.$index++];

But if I make that change, a unit test breaks.

BUT...

def test_sinkstate_list():
        # This used to fail
        a = range(5)
        b = iter(a)
        # print "list of b", list(b)
        # print next(b)

        assert list(b)==range(5)
        a.extend(range(5, 10))
        assert list(b) == []

BUT...that particular unit test seems to test a Python 2 behaviour. Python 3 agrees with my patch, not with Python 2.

SO...I propose that we should go with the patch and change the unit test but I don't know the Skulpt project's policy on these issues.

**Comment by albertjan** [comment]
@prescod did you see my PR? Am I missing something or have I accidentally found the way to fix this 😬
